src/ServicePulse.Host/app/js/views/endpoint_details/endpoint_details.controller.js

```diff
@@ -4,41 +4,73 @@
     function controller(
         $scope,
         $routeParams,
+        $location,
         toastService,
         serviceControlService,
         monitoringService) {
 
         $scope.endpointName = $routeParams.endpointName;
         $scope.sourceIndex = $routeParams.sourceIndex;
         $scope.loading = true;
+        var subscription;
 
-        var subscription = monitoringService.endpointDetails($routeParams.endpointName, $routeParams.sourceIndex).subscribe(function (endpointInstances) {
-            if (endpointInstances.error) {
-                // show warning
-                toastService.showWarning('Could not load endpoint details');
-            } else {
-                $scope.endpointInstances = endpointInstances;
-                $scope.loading = false;
+        $scope.periods = [
```

Comment:
How about we define it as contestant and reuse it in endpoint monitoring and endpoint details?